### PR TITLE
PYTHON-3480 Multiple Build Failures in 3.13 Due to Running Newer Servers on Amazon Linux

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -1436,11 +1436,29 @@ tasks:
...
     - name: "mod-wsgi-replica-set"
+      tags: ["mod_wsgi"]
+      commands:
+        - func: "bootstrap mongo-orchestration"
+          vars:
+            VERSION: "5.0"
+            TOPOLOGY: "replica_set"
+        - func: "run mod_wsgi tests"
+
+    - name: "mod-wsgi-standalone-latest"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:1452]: I think we can stop test latest altogether on the failing platforms in this PR. It's not worth trying to migrate.

Comment (blink1073) [reply .evergreen/config.yml]: Done.  Load balancer tests are now [passing](https://spruce.mongodb.com/version/63517dcd3627e05bababdce9/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

Comment (blink1073) [reply .evergreen/config.yml]: Also mod_wsgi with 5.0

File: .evergreen/config.yml [hunk 2]
```diff
@@ -2931,7 +2931,7 @@ buildvariants:
   tasks:
      - name: "test-enterprise-auth"
 
-- matrix_name: "tests-mod-wsgi"
+- matrix_name: "tests-mod-wsgi-aws"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2934]: Let's revert the name change to avoid loosing task history.

File: .evergreen/config.yml [hunk 3]
```diff
@@ -3111,17 +3111,28 @@ buildvariants:
     - name: "aws-auth-test-5.0"
     - name: "aws-auth-test-latest"
 
-- matrix_name: "load-balancer"
+- matrix_name: "load-balancer-awslinux"
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:3114]: Let's revert the name change to avoid loosing task history.

**Review by ShaneHarvey — COMMENTED** [review]
two minor comments otherwise lgtm

File: .evergreen/config.yml (hunk 2 above)
Comment (blink1073) [reply .evergreen/config.yml]: Done

File: .evergreen/config.yml (hunk 3 above)
Comment (blink1073) [reply .evergreen/config.yml]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**